### PR TITLE
fixed #6596 Missing field name tooltip

### LINKED ISSUE #6596: Missing field name tooltip
## Scope & Context

Object Record Detail Page

## Current behavior

I added some fields to my custom investor model. The field names are a bit too long to be fully displayed on the object record detail page. Because of this, I couldn't distinguish between the "Initial Investment (min)" and "Initial Investment (max)" fields and ended up entering the wrong values.

Bildschirmfoto 2024-08-09 um 20 46 17

## Expected behavior

There should be a tooltip or hover text that appears when hovering over the field name, allowing the full field name to be read. Additionally, it would be very helpful if fields could be sorted manually rather than alphabetically.

### PR DESCRIPTION
**Pull request by Sandheep-OSC**
Fixes for #6596 
image

### CODE REVIEW

**Review by lucasbordeau — COMMENTED** [review]
This solution is not right because here the AppTooltip is for BoardCard that don't have labels to show. We don't want to modify this line.

Instead we should create a component RecordInlineCellLabel that will be used above this line for the case where label is defined and showLabel is true.

In this component we should create a new component for that : `EllipsisTextWithTooltip`, to let `EllipsisDisplay` intact (because it is used everywhere and it has performance impacts)

In this new component `EllipsisTextWithTooltip` we should check if there is an overflow of text, using `getBoundingClientRect()` on the elements (in a useEffect)

Compute isOverflowing with `getBoundingClientRect()` and allow a tooltip to be shown on mouse hover if isOverflowing is true, it should be the same as the current AppTooltip for the `!showLabel` part in RecordInlineCellContainer.

This way we'll have a tooltip for overflowing labels only.

**Comment by Sandheep-OSC** [comment]
Hey @lucasbordeau sorry for so long inactive hold, had some health issue. So should i create `EllipsisTextWithTooltip` component as an internal component of `RecordInlineCellContainer` or a new component inside `twenty-ui` package?

**Comment by Sandheep-OSC** [comment]
@lucasbordeau i tried using `offsetWidth` and `scrollWidth` from `getBoundingClientRect` to check if there is an overflow or not but both of these values are always same as the `maxWidth` which we pass to `EllipsisDisplay`. Due to this, the isOverflow is always getting set as false. Here is my code

```typescript
import React, { useEffect, useRef, useState } from 'react';
import { EllipsisDisplay } from './EllipsisDisplay';
import { AppTooltip, TooltipDelay } from 'twenty-ui';

type EllipsisDisplayProps = React.ComponentProps<typeof EllipsisDisplay>;

type EllipsisTextWithTooltipProps = EllipsisDisplayProps & {
  tooltip: string;
  labelId: string;
};

const EllipsisTextWithTooltip = (props: EllipsisTextWithTooltipProps) => {
  const textRef = useRef<HTMLDivElement>(null);
  const [isOverflowing, setIsOverflowing] = useState(false);
  const [showTooltip, setShowTooltip] = useState(false);

  useEffect(() => {
    const checkOverflow = () => {
      if (textRef.current !== null) {
        const { offsetWidth, scrollWidth } = textRef.current;
        setIsOverflowing(scrollWidth >= offsetWidth);
      }
    };

    checkOverflow();
  }, [props.children, props.maxWidth]); // Added props.maxWidth to dependency array

  const handleMouseEnter = () => {
    if (isOverflowing) {
      setShowTooltip(true);
    }
  };

  const handleMouseLeave = () => {
    setShowTooltip(false);
  };

  return (
    <div
      ref={textRef}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <EllipsisDisplay maxWidth={props.maxWidth} className={props.className}>
        {props.children}
      </EllipsisDisplay>
      {showTooltip && isOverflowing && (
        <AppTooltip
          anchorSelect={`#${props.labelId}`}
          content={props.tooltip}
          clickable
          noArrow
          place="bottom"
          positionStrategy="fixed"
          delay={TooltipDelay.shortDelay}
        />
      )}
    </div>
  );
};

export default EllipsisTextWithTooltip;
```

**Comment by lucasbordeau** [comment]
I used your solution to do this, it works, could you please implement this and push the changes : 

```tsx
import { StyledEllipsisDisplay } from '@/ui/field/display/components/EllipsisDisplay';
import { useRef, useState } from 'react';
import { AppTooltip, TooltipDelay } from 'twenty-ui';

type EllipsisTextWithTooltipProps = {
  labelId: string;
  text: string;
};

export const EllipsisTextWithTooltip = ({
  labelId,
  text,
}: EllipsisTextWithTooltipProps) => {
  const textRef = useRef<HTMLDivElement>(null);

  const [showTooltip, setShowTooltip] = useState(false);

  const checkIsOverflowing = () => {
    if (textRef.current !== null) {
      const { offsetWidth, scrollWidth } = textRef.current;

      const isOverflowing = scrollWidth > offsetWidth;

      return isOverflowing;
    }

    return false;
  };

  const handleMouseEnter = () => {
    const isOverflowing = checkIsOverflowing();

    if (isOverflowing) {
      setShowTooltip(true);
    }
  };

  const handleMouseLeave = () => {
    setShowTooltip(false);
  };

  return (
    <>
      <div onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
        <StyledEllipsisDisplay ref={textRef} id={labelId}>
          {text}
        </StyledEllipsisDisplay>
        {showTooltip && (
          <AppTooltip
            anchorSelect={`#${labelId}`}
            content={text}
            isOpen
            clickable
            noArrow
            place="bottom"
            positionStrategy="fixed"
            delay={TooltipDelay.shortDelay}
          />
        )}
      </div>
    </>
  );
};
```

**Comment by Sandheep-OSC** [comment]
Made the changes and pushed the code @lucasbordeau

File: packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCellContainer.tsx [hunk 1]
```diff
@@ -7,10 +7,11 @@ import { FieldContext } from '@/object-record/record-field/contexts/FieldContext
 import { useFieldFocus } from '@/object-record/record-field/hooks/useFieldFocus';
 import { RecordInlineCellValue } from '@/object-record/record-inline-cell/components/RecordInlineCellValue';
 import { getRecordFieldInputId } from '@/object-record/utils/getRecordFieldInputId';
-import { EllipsisDisplay } from '@/ui/field/display/components/EllipsisDisplay';
+// import { EllipsisDisplay } from '@/ui/field/display/components/EllipsisDisplay';
```
Comment (lucasbordeau) [inline packages/twenty-front/src/modules/object-record/record-inline-cell/components/RecordInlineCellContainer.tsx:10]: Remove dead code

**Review by lucasbordeau — COMMENTED** [review]
Could you please change the branch on which this PR is made, it's very hard for us to review when you push from your main branch.

**Comment by Sandheep-OSC** [comment]
> Could you please change the branch on which this PR is made, it's very hard for us to review when you push from your main branch.

Created a new pr from a new branch @lucasbordeau. Should i close this pr?
https://github.com/twentyhq/twenty/pull/6908

**Pull request closed without merging**